### PR TITLE
[DCOS-44704] Implement OfferOutcome debug endpoint

### PR DESCRIPTION
**Pull request by kvish**
currently going for parity with the marathon v2/debug/offers endpoint 

will add more information here as we learn this in debuggability

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java [hunk 1]
```diff
@@ -87,6 +90,7 @@ public OfferEvaluator(
       FrameworkStore frameworkStore,
       StateStore stateStore,
       Optional<OfferOutcomeTracker> offerOutcomeTracker,
+      Optional<OfferOutcomeTrackerV2> offerOutcomeTrackerV2,
```
Comment (takirala) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java]: (not sure if this has already been discussed) Considering that the current debug endpoint is for debugging only, i think we can just remove this completely in favor of the new endpoint(s). It's not like the user would be losing some information as the enriched information is present at the new endpoint anyway. wdyt ?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/debug/OfferOutcomeTrackerV2.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      if (this.failureReasons.containsKey(failureReason)) {
+        int oldVal = this.failureReasons.get(failureReason);
+        this.failureReasons.put(failureReason, ++oldVal);
+      } else {
+        this.failureReasons.put(failureReason, 1);
+      }
+    }
+
+    public void addFailureAgent(String agentId) {
+      if (this.rejectedAgents.containsKey(agentId)) {
```
[the remark this answers is not in the document]
Comment (kvish) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/debug/OfferOutcomeTrackerV2.java]: thanks! was looking for a cleaner way to do this

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java (hunk 1 above)
Comment (kvish) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java]: ya I agree, however according to the current plan we will completely cut the v1 endpoint once the UI is using v2, and we have fine tuned the v2 endpoint after some trial. This isnt an issue for now, im for keeping it and dropping v1 in a separate PR.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/debug/OfferOutcomeTrackerV2.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    public JSONArray toJson() {
+      JSONArray displayOutcomes = new JSONArray();
+      this.outcomes.forEach(offerOutcome -> {
+        JSONObject outcome = new JSONObject();
+        outcome.put("timestamp", offerOutcome.getTimestamp())
+            .put("pod-instance-name", offerOutcome.getPodInstanceName())
+            .put("outcome", offerOutcome.pass() ? "pass" : "fail")
+            .put("explanation", offerOutcome.getOutcomeDetails())
+            .put("offer", offerOutcome.getOffer());
+        displayOutcomes.put(outcome);
```
[the remark this answers is not in the document]
Comment (kvish) [reply sdk/scheduler/src/main/java/com/mesosphere/sdk/debug/OfferOutcomeTrackerV2.java]: sure here is a sample:
`"rejectedAgents": {
"884f88ad-968b-41e6-82e5-c2bb3ad97bde-S1": 1
},
"failureReasons": {
"PlacementRuleConstraint": 1
},`

**Pull request merged**